### DISCUSSION TITLE
Upgrading from 0.8.3 to 0.9.0 - Failed to update ephemeral runner set with removed finalizer

### DISCUSSION THREAD

**Discussion by rekha-prakash-maersk** [actions/actions-runner-controller#3421] [Questions] [answered]
Hi, 

I tried upgrading from 0.8.3 to 0.9.0 using FluxCD and facing below error, I tried to deploy the config in a new namespace with new runner group as a workaround. But still its referring to the old namespace and old runner scaleset name.

> 2024-04-09T14:31:49Z    INFO    EphemeralRunnerSet    Deleting resources    {"ephemeralrunnerset": {"name":"self-host-kcbwv","namespace":"arc"}}                                          
││ 2024-04-09T14:31:49Z    INFO    EphemeralRunnerSet    Actual Ephemeral runner counts    {"ephemeralrunnerset": {"name":"self-host-kcbwv","namespace":"arc"}, "count": 0}                  
││ 2024-04-09T14:31:49Z    INFO    EphemeralRunnerSet    All ephemeral runners are deleted    {"ephemeralrunnerset": {"name":"self-host-kcbwv","namespace":"arc}}                           
││ 2024-04-09T14:31:49Z    INFO    EphemeralRunnerSet    Removing finalizer    {"ephemeralrunnerset": {"name":"self-host-kcbwv","namespace":"arc"}}                                          
││ 2024-04-09T14:31:49Z    ERROR    EphemeralRunnerSet    Failed to update ephemeral runner set with removed finalizer	{"ephemeralrunnerset": {"name":"self-host-kcbwv","namespace":"arc"}, "error": "EphemeralRunnerSet.actions.github.com \"ubuntu-latest-kcbwv\" is invalid: spec.patchID: Required value"

**Comment by rekha-prakash-maersk** [comment]
I had to delete the old namespace by setting finalizer as null in the namespace spec as the finalizer was not allowing to delete it. But I am clueless why the controller deployed in new namespace still referring the old scaleset name and namespaces ?

**Comment by dmaclaury** [comment, marked as answer]
I was running into this issue as well. I had to manually edit the old `EphemeralRunnerSet`'s and remove the finalizer, as well as add `spec.patchID: 1`. It looks like this field was added in the `0.9.0` release.
